Test on chain sync (#2050)
valory-xyz/open-autonomy: docs/advanced_reference/commands/autonomy_mint.md

    - `GOERLI_CHAIN_RPC` : RPC endpoint for the Görli testnet chain.

`--use-custom-chain`

> : Use the custom-chain chain profile to interact with the Autonolas Protocol registry contracts. This option requires that you define the following environment variables (see the {{ autonolas_protocol }} and [On Chain Contract Addresses](../on_chain_addresses.md) documentation for more information):

Reviewer: ```suggestion
: Use the custom-chain chain profile to interact with the Autonolas Protocol registry contracts. To define the custom-chain profile you must define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):
```

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

> : Use the custom-chain profile to interact with the {{ autonolas_protocol }} registry contracts. This profile requires that you define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):

Reviewer: ```suggestion
: Use the custom-chain profile to interact with the Autonolas Protocol registry contracts. This profile requires that you define some parameters and [contract addresses](../on_chain_addresses.md) as environment variables (see also the {{ autonolas_protocol }} documentation for more information):
```

    - `CUSTOM_CHAIN_RPC` : RPC endpoint for the custom chain.
    - `CUSTOM_CHAIN_ID` : chain ID.
    - `CUSTOM_COMPONENT_REGISTRY_ADDRESS` : Custom Component Registry contract address.
    - `CUSTOM_AGENT_REGISTRY_ADDRESS` : Custom Agent Registry contract address.
    - `CUSTOM_REGISTRIES_MANAGER_ADDRESS` : Custom Registries Manager contract address.

>     - `CUSTOM_SERVICE_MANAGER_ADDRESS` : Custom Service Manager contract address.

Author: The manager with the token util is only deployed on L1 chains eg ethereum and goerli and we provide these 2 configurations with the framework. Env vars are only used for L2 chains so no need for `SERVICE_MANAGER_TOKEN_ADDRESS `

>     - `CUSTOM_SERVICE_MANAGER_ADDRESS` : Custom Service Manager contract address.

Reviewer 2: if this is only for L2s, why we have `CUSTOM_COMPONENT_REGISTRY_ADDRESS`, `CUSTOM_AGENT_REGISTRY_ADDRESS`,`CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS` ?

>     - `CUSTOM_SERVICE_MANAGER_ADDRESS` : Custom Service Manager contract address.

Author: Just in case if we deploy on some other L1 chain so the users won't have to wait until the next release

>     - `CUSTOM_SERVICE_MANAGER_ADDRESS` : Custom Service Manager contract address.

Reviewer 2: Then **SERVICE_MANAGER_TOKEN_ADDRESS **  should be added as well. Agreed?

>     - `CUSTOM_SERVICE_MANAGER_ADDRESS` : Custom Service Manager contract address.

Reviewer 2: I see where this come from now. You call on both L1 and L2 service manager. Maybe for easier understanding would be better to explicitly show the difference

    - `CUSTOM_SERVICE_REGISTRY_ADDRESS` : Custom Service Registry contract address.
    - `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS` : Custom Gnosis Safe multisig contract address.
    - `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` : Custom Gnosis Safe Same Address Multisig address.
    - `CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS` : Custom Service Registry Token Utility address.
    - `CUSTOM_MULTISEND_ADDRESS` : Custom Multisend address.

!!! note

>     For L2 chains you will only require to set `CUSTOM_SERVICE_REGISTRY_ADDRESS`, `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS`, `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` and `CUSTOM_MULTISEND_ADDRESS`

Author: Addressed Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

> !!! note
> 
>     For L2 chains you will only require to set `CUSTOM_SERVICE_MANAGER_ADDRESS`, `CUSTOM_SERVICE_REGISTRY_ADDRESS`, `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS`, `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` and `CUSTOM_MULTISEND_ADDRESS`

Reviewer: ```suggestion
    !!! note

        For L2 chains you are only required to set

        - `CUSTOM_SERVICE_MANAGER_ADDRESS`,
        - `CUSTOM_SERVICE_REGISTRY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_PROXY_FACTORY_ADDRESS`,
        - `CUSTOM_GNOSIS_SAFE_SAME_ADDRESS_MULTISIG_ADDRESS` and
        - `CUSTOM_MULTISEND_ADDRESS`.
```

`--owner OWNER_ADDRESS`
: Owner address of the package.

`--update TOKEN_ID`

> : Update the existing mint with the latest package hash.

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

`--owner OWNER_ADDRESS`
: Owner address of the package.

`--update TOKEN_ID`

> : Update the existing minted token with the latest package hash.

Reviewer: ```suggestion
: Update the already minted  protocol, contract, connection or skill with on-chain TOKEN_ID with the current package hash.
```

`--owner OWNER_ADDRESS`
: Owner address of the package.

`--update TOKEN_ID`

> : Update the existing mint with the latest package hash.

Reviewer 2: same suggestion of previous comment

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

`--threshold`
: Threshold for the minimum number of agents required to run the service. The threshold has to be at least $\lceil(2N + 1) / 3\rceil$, where $N$ is total number of the agents in the service.

`--token ERC20_TOKEN_ADDRESS`

> : Use the ERC20 token for bonding.

Reviewer 2: ```suggestion
: Use the ERC20 token for securing autonomous services.
```

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

`--update TOKEN_ID`

> : Update the existing mint with the latest package hash.

Reviewer 2: same suggestion of previous comment

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

`--threshold`
: Threshold for the minimum number of agents required to run the service. The threshold has to be at least $\lceil(2N + 1) / 3\rceil$, where $N$ is total number of the agents in the service.

`--token ERC20_TOKEN_ADDRESS`

> : Use the ERC20 token for securing autonomous services.

Reviewer: ```suggestion
: ERC20 token for securing the service.
```

!!! note

    You can specify more than one type of canonical agent in a service by appropriately defining the triplets `--agent-id`, `--number-of-slots` and `--cost-of-bond` for each canonical agent ID.

> You can also use a custom ERC20 token as bonding token for a service, to do this use `--token` flag to provide the address to the token of your choice:

Reviewer 2: ```suggestion
You can also use a custom ERC20 token as token for secure the autonomous service, to do this use `--token` flag to provide the address to the token of your choice:
```

Author: Addressed [c241c59](https://github.com/valory-xyz/open-autonomy/pull/2050/commits/c241c59d71cdbcfc957398e1e6eb69eaef5daab1)

!!! note

    You can specify more than one type of canonical agent in a service by appropriately defining the triplets `--agent-id`, `--number-of-slots` and `--cost-of-bond` for each canonical agent ID.

> You can also use a custom ERC20 token as token for secure the autonomous service, to do this use `--token` flag to provide the address to the token of your choice:

Reviewer: ```suggestion
You can also use a custom ERC20 token as token to secure the service. Use the `--token` flag to provide the address of the token of your choice:
```

```bash
autonomy mint --use-ethereum service --key my_key.txt --nft <nft_ipfs_hash_or_image_path> --owner <owner_address> --agent-id 3 --number-of-slots 4 --cost-of-bond 10000000000000000 --threshold 3 ./packages/valory/services/hello_world --token <erc20_token_address>
```

> !!! note
> 
>     When minting a service if an ERC20 token was used for bonding, you'll also have to use the same token when activating the service and registering the agent instances.

Reviewer: ```suggestion
!!! warning "Important"

    If you have minted a service using a custom ERC20 token, then you have to use the same token to register the agent instances.
```
